File: 2733-neither-minimum-nor-maximum/2733-neither-minimum-nor-maximum.cpp
class Solution {
public:
int findNonMinOrMax(vector<int>& nums) {
    auto min = *min_element(nums.begin(), nums.end()), max = *max_element(nums.begin(), nums.end());

    nums.erase(remove(nums.begin(), nums.end(), min), nums.end());
    nums.erase(remove(nums.begin(), nums.end(), max), nums.end());

    try {
        return nums.at(0);
    } catch (out_of_range e) { return -1; }
}
};

File: 2733-neither-minimum-nor-maximum/2733-neither-minimum-nor-maximum.kt
class Solution {
    fun findNonMinOrMax(nums: IntArray): Int {
        val minIndex = nums.indexOf(nums.min() ?: 0)
        val maxIndex = nums.indexOf(nums.max() ?: 0)
        
        val num = nums.apply { 
            set(minIndex, Int.MIN_VALUE)
            set(maxIndex, Int.MIN_VALUE)
        }.max() ?: Int.MIN_VALUE
        
        return if (num == Int.MIN_VALUE) -1 else num
    }
}